src/options.cc

```diff
@@ -67,6 +70,29 @@ bool Options::addDefaultHandler(std::shared_ptr<NginxConfigStatement> handler_co
     return addHandler(type, path, *handler_config->child_block_);
 }
 
+bool Options::addAuthentication(std::shared_ptr<NginxConfigStatement> auth_config) {
+    if (auth_config->tokens_.size() != REALM_SIZE) {
+        std::cerr << "A path has not been specified for the realm.\n";
+        return false;
+    }
+
+    auth = new Authentication();
+    std::map<std::string, std::string> AuthMap;
+    for (auto& statements : auth_config->child_block_->statements_) {
+        if (statements->tokens_.size() != REALM_CRED_SIZE) {
+            std::cerr << "Not a valid credential statement. Need " << 
+                REALM_CRED_SIZE << " tokens.\n";
+            return false;
+        }
+        if(statements->tokens_[0] == "credential") {
+            std::string username = statements->tokens_[1];
+            std::string pass = statements->tokens_[2];
+            AuthMap[username] = pass;
+        }
+    }
+    auth->addRealm(new AuthenticationRealm(auth_config->tokens_[VAL], AuthMap));
+    return true;
```

Comment:
So 2 things:
1. I don't think you ever set `options.auth = auth`, so it won't ever see any of the configured settings.
2. The `Authentication` is the global handler of all realms, so we probably want to initialize `options.auth`
outside of here, then just call `addRealm` for each realm block in the config. I guess this method might be better called `Options::addAuthenticationRealm`